### DISCUSSION TITLE
Easy way to provision VMs programatically *exactly* like through the web UI?

### DISCUSSION THREAD

**Discussion by alexpdp7** [cockpit-project/cockpit-machines#1531] [Q&A] [answered]
I am running a small POC. I've been able to deploy a Ubuntu 22.04 from a cloud image without much friction through the web interface.

I'm wondering if there's a nice way to extract the same `virt-install` command that the UI uses, so I can alternatively create VMs that behave exactly the same without going through the web interface.

**Comment by jelly** [comment, marked as answer]
We don't really expose this and I don't think we should, but you can find the virt-install line here https://github.com/cockpit-project/cockpit-machines/blob/9982a32eb3808e9061f2f0a502f136086a89a758/src/scripts/install_machine.py#L192

Not sure if there is a tool which can convert the virt xml to a virt-install command.

**Reply by alexpdp7** [reply]
That's exactly what I wanted- sorry, I was looking at the wrong places for that, so I couldn't find it. Thanks!

**Reply by jelly** [reply]
No problem, happy to help!
